UI: VAULT-16203 tidy status page

kiannaquach opened the pull request:
**Description**
- Cross-cluster operation section is only shown for enterprise customers
- Added image for modal
- Used the `HDS::Alert` for the tidy status state banner
- Add timestamp for the tidy status banner
- Paired with Chelsea on having a background refresh every 2 seconds.
- **TODO:** Custom adapter for cancel tidy to follow after Claire merges her work with the tidy forms. 
- **Nice to have:** Add acceptance tests for the tidy status page

<img width="716" alt="Screenshot 2023-05-17 at 3 47 53 PM" src="https://github.com/hashicorp/vault/assets/30884335/92780f5f-af1d-4bb4-94a1-15f76ca46676">
<img width="735" alt="Screenshot 2023-05-17 at 3 28 28 PM" src="https://github.com/hashicorp/vault/assets/30884335/d37c662d-0dd0-49d5-8017-ed3cca7ab923">
<img width="740" alt="Screenshot 2023-05-17 at 3 25 49 PM" src="https://github.com/hashicorp/vault/assets/30884335/a4111c13-1806-4641-88d6-cea8e3c70079">

when tidy has never been performed, the natural empty state is:
<img width="755" alt="Screenshot 2023-05-17 at 5 38 30 PM" src="https://github.com/hashicorp/vault/assets/30884335/e0fed817-71b5-49bc-bbc5-19043637b169">

**From the conversation with Claire:**
After team discussion, there are too many assumptions we have to make regarding whether a tidy operation is inactive because there has NEVER been one, or if it's just the period between tidying certificates.

As a follow-on we'd like to discuss with the backend if they can add a parameter that indicates tidying has never been performed - or if there are existing params we can draw that conclusion from

**Updated background refresh behavior:**
![radio](https://github.com/hashicorp/vault/assets/30884335/f7c96516-50dc-4a04-9ae6-5cd43e5e7cf9)

hellobontempo commented on ui/lib/pki/addon/components/page/pki-tidy-status.hbs:
```diff
@@ -28,9 +18,87 @@
   </ToolbarActions>
 </Toolbar>
 
+<Hds::Alert
```
😍

hellobontempo commented on ui/lib/pki/addon/components/page/pki-tidy-status.js:
```diff
@@ -0,0 +1,118 @@
+    }
+  }
+
+  get tidyState() {
```
nit: It might be nice to move this above `tidyStateAlertBanner` so it's easier follow that getter and where `tidyState` is coming from

hellobontempo commented on ui/lib/pki/addon/components/page/pki-tidy-status.js:
do we want this file to be typescript?

kiannaquach replied:
yes, that's another TODO :)

hellobontempo commented on ui/lib/pki/package.json:
```diff
@@ -8,6 +8,7 @@
     "ember-cli-babel": "*",
     "ember-cli-htmlbars": "*",
     "ember-cli-typescript": "*",
+    "@hashicorp/design-system-components": "2.3.1",
```
per Jordan's conversation in the dev channel - do we want this to have a `*` so the main app manages the version?

hellobontempo commented on ui/tests/acceptance/pki/pki-engine-workflow-test.js:
```diff
@@ -449,7 +449,7 @@ module('Acceptance | pki workflow', function (hooks) {
       await visit(`/vault/secrets/${this.mountPath}/pki/configuration`);
       await click(SELECTORS.tidyTab);
-      assert.strictEqual(currentURL(), `/vault/secrets/${this.mountPath}/pki/configuration/tidy`);
+      assert.strictEqual(currentURL(), `/vault/secrets/${this.mountPath}/pki/tidy`);
```
thank you for fixing!

hellobontempo commented on ui/tests/integration/components/pki/page/pki-tidy-status-test.js:
```diff
@@ -0,0 +1,121 @@
+import { setupMirage } from 'ember-cli-mirage/test-support';
+
+const SELECTORS = {
+  hdsAlertTitle: '.hds-alert .hds-alert__title',
```
I wonder if using class selectors here could cause problems later when the HDS banner is used more throughout the app? We've had banner-selector issues before... 🤔  Thought son adding a test selector to the hds component in the `pki-tidy-status.hbs` template?

hellobontempo commented on ui/lib/pki/addon/components/page/pki-tidy-status.hbs:
```diff
@@ -28,9 +18,87 @@
+</Hds::Alert>
+
+{{#each this.generalSectionFields as |attr|}}
+  {{#if (or (eq attr "time_started") (eq attr "time_finished") (eq attr "last_auto_tidy_finished"))}}
```
Mostly as a handlebar FYI you could replace the conditional logic here leverage the `@formatDate` arg to do something like 

```js
<InfoTableRow
  @label={{humanize (dasherize attr)}}
  @formatDate={{if (includes attr (array "time_started" "time_finished" "last_auto_tidy_finished")) "MMM dd, yyyy hh:mm:ss a"}}
  @value={{get @tidyStatus attr}}
  @alwaysRender={{true}}
    />

hellobontempo commented on ui/lib/pki/addon/components/page/pki-tidy-status.hbs:
```diff
@@ -28,9 +18,87 @@
+      @alwaysRender={{true}}
+    />
+  {{else}}
+    <InfoTableRow @label={{humanize (dasherize attr)}} @value={{(get @tidyStatus attr)}} @alwaysRender={{true}} />
```
also fyi don't need the extra `()` here - same comment for line 95 below! 
```suggestion
    <InfoTableRow @label={{humanize (dasherize attr)}} @value={{get @tidyStatus attr}} @alwaysRender={{true}} />
```

hellobontempo reviewed (approved):
Great work! A few minor comments, nothing blocking!

hellobontempo reviewed (approved):
Great work! A few minor comments, nothing blocking!

kiannaquach commented:
@hellobontempo thanks for the feedback! Could you take another look at the PR with the updated changes? I just addressed them and converted the component file to TS since it didn't take too long!

kiannaquach commented on ui/tests/acceptance/pki/pki-engine-workflow-test.js:
```diff
@@ -444,19 +444,6 @@ module('Acceptance | pki workflow', function (hooks) {
         .hasValue('Hashicorp Test', 'form prefilled with parent issuer cn');
     });
-
-    test('it returns to the configuration page after submit', async function (assert) {
```
this test isn't relevant anymore since we have a tidy route the functionality is different.

hellobontempo commented on ui/lib/pki/addon/components/page/pki-tidy-status.ts:
```diff
@@ -3,16 +3,146 @@
+
+interface Args {
+  autoTidyConfig: PkiTidyModel;
+  tidyStatus: {
```
having these types here is nice!

hellobontempo reviewed (approved):
Looks good!
